### ISSUE TITLE
S507 does not trigger when using class inheritance

### ISSUE THREAD

**Issue by schnow265** [astral-sh/ruff#21407]
### Summary

When creating a class which inherits from `paramiko.SSHClient` and then using `self.set_missing_host_key_policy(paramiko.AutoAddPolicy)`, the linter rule is not triggered. The same applies when creating an instance of the custom class and calling the function there.

I have included an example:

```python
import paramiko

ssh_client = paramiko.SSHClient()
ssh_client.set_missing_host_key_policy(paramiko.AutoAddPolicy)  # triggers rule


class Test(paramiko.SSHClient):
    def __init__(self):
        super().__init__()
        self.set_missing_host_key_policy(paramiko.AutoAddPolicy)  # does not trigger rule

client = Test()
client.set_missing_host_key_policy(paramiko.AutoAddPolicy)  # also doesn't trigger rule
```

You can check it by pasting my python snippet into a file and then using the following command:

```shell
ruff check --select S507 ./path/to/file.py
```

### Version

ruff 0.14.4

**Comment by ntBre** [comment]
Thanks for the report! I think we could try to resolve base classes within the same file at least, even if multi-file analysis is still unavailable.

**Comment by dataCenter430** [comment]
Hi, @schnow265 thanks for reporting this.
I'd love to work on this.
